### ISSUE TITLE
Status messages on axios errors are sometimes incomplete or misleading

### ISSUE THREAD

**Issue by mrentsch65** [axios/axios#6222]
### Describe the bug

See the following two cases:
1) The status message "Invalid URL" does not match the reason displayed by Browser "Mixed Content":
![image](https://github.com/axios/axios/assets/50681355/a1703708-1bca-4177-b1a1-c45c31e0ca4b)
2) Although received, there is no status code 200 forwarded by axios:
![image](https://github.com/axios/axios/assets/50681355/57938726-3844-4286-b1a8-603c0156498b)
(in the case above, the reponse header returns "Content-Type:  text/html; charset=UTF-8")

### Expected behavior

1) The status Message should be "Mixed Content" or "Unsecure Content", not "InvalidURL".
2) The HTTP status code "200" should be forwarded, because it was received.

**Comment by DigitalBrainJS** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Comment by mrentsch65** [comment]
> https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
Yes, the reason is surely CORS. The problem is that axios doesn't report that in a comprehensible way to the calling application, not even forwarding the received HTTP status code. 
Is there a technical reason for it?

**Comment by DigitalBrainJS** [comment]
> Is there a technical reason for it?

CORS specification. The short answer is that the JS environment is not allowed to see the real cause of the error. Here's a pretty good answer that explains in detail why this is so https://stackoverflow.com/a/19325710/11046238

**Comment by jasonsaayman** [comment]
closing as a good solution / explaination was provided
